File: src/main/java/com/addressbook/dto/TableDataDto.java
package com.addressbook.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class TableDataDto<T> {

    private List<T> data;
    private Integer totalDataSize;

    public TableDataDto(List<T> data, Integer totalDataSize) {
        this.data = data;
        this.totalDataSize = totalDataSize;
    }
}

File: addressbook-main/src/main/kotlin/com/addressbook/dto/TableDataDto.kt
package com.addressbook.dto

class TableDataDto<T> constructor(data: List<T>?, totalDataSize: Int?) {

    var data: List<T>? = null
    var totalDataSize: Int? = null

    init {
        this.data = data
        this.totalDataSize = totalDataSize
    }

    constructor(data: List<T>?) : this(data, data?.size)
}
